### ISSUE TITLE
Literal["some string"] incorrectly interepreted as a forward reference

### ISSUE THREAD

**Issue by CallumJHays** [microsoft/pyright#1420]
![image](https://user-images.githubusercontent.com/13251689/105797840-885a5f80-5fdc-11eb-924b-5fc66f88dfc7.png)

According to [PEP586, this behaviour is incorrect:](https://www.python.org/dev/peps/pep-0586/#literals-enums-and-forward-references)
> In cases like this (`Literal["Color.RED"]`), we always assume the user meant to construct a literal string. If the user wants a forward reference, they must wrap the entire literal type in a string -- e.g. `"Literal[Color.RED]"`.

I'm using Pylance version: `v2021.1.2` with:
```json
"python.analysis.typeCheckingMode": "strict",
```

**Comment by erictraut** [comment]
The implementation follows PEP 586. I'm not able to repro the problem you're reporting.

Here's what I see (as expected) given your code snippet:
![Screen Shot 2021-01-25 at 9 08 10 PM](https://user-images.githubusercontent.com/7040122/105803050-6ff13300-5f51-11eb-8381-11ad19aa3a70.png)

By any chance, are you overriding the standard `typing_extensions.pyi` file with a custom type stub? If you command-click on `Literal`, where does it take you? I'm at a loss to come up with any other explanations for what you're seeing.

**Comment by CallumJHays** [comment]
Hi, thanks for your assistance :), I've fixed the issue.

I'm using a conda environment with python 3.6, `typing_extensions` pointed to `/home/cal/.conda/envs/bdsim.old/lib/python3.6/site-packages/typing_extensions.py`. `pip show`s the version of the package to be `Version: 3.7.4.3`.

I had added `"/home/cal/.conda/envs/bdsim.old/lib/python3.6/site-packages"` to my `"python.analysis.extraPaths"` in `settings.json` on top of the pythonPath locking onto the correct (conda) python binary. I did this to resolve some import errors that don't appear to be occurring anymore (I'm sure I had the packages installed), so not sure what's up with that.

Anyway, removing that extra path from my settings fixed the issue.

I'm not 100% sure on the root cause of the issue. Is the version of `typing-extensions` I'm using not conformant to the latest PEP 586? Looks like it makes no difference at runtime anyhow.

Thanks!
